test/functional/cursor_tests.js

```diff
@@ -4423,4 +4423,79 @@ describe('Cursor', function() {
       });
     });
   });
+
+  function testTransformStream(client, configuration, dbName, transformFunc, expectedSet, done) {
+    client.connect(function(err, client) {
+      const db = client.db(configuration.db);
+      let collection, cursor;
+      const docs = [
+        { _id: 0, a: { b: 1, c: 0 } },
+        { _id: 1, a: { b: 1, c: 0 } },
+        { _id: 2, a: { b: 1, c: 0 } }
+      ];
+      const resultSet = new Set();
+      const transformParam = transformFunc != null ? { transform: transformFunc } : null;
+
+      Promise.resolve()
+        .then(() => db.createCollection(dbName))
+        .then(() => (collection = db.collection(dbName)))
+        .then(() => collection.insertMany(docs))
+        .then(() => collection.find())
+        .then(_cursor => (cursor = _cursor))
+        .then(() => cursor.transformStream(transformParam))
+        .then(stream => {
+          stream.on('data', function(doc) {
+            resultSet.add(doc);
+          });
+
+          stream.once('end', function() {
+            expect(resultSet).to.deep.equal(expectedSet);
+            cursor.close(() => client.close(() => done()));
```

Comment:
it looks like this code can be defined commonly as the `cleanup` handler or something, it's called three times in this test.